server/channels/api4/channel_category_test.go

```diff
@@ -550,6 +551,734 @@ func TestUpdateCategoriesForTeamForUser(t *testing.T) {
+	})
+
+	t.Run("should filter out invalid channels", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		channels := model.ChannelList{
+			th.BasicChannel,
+		}
+
+		channelIds := []string{
+			th.BasicChannel.Id,
+			"invalid_channel_id",
+		}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, channelIds, channels)
+		require.Len(t, filtered, 1)
+		require.Contains(t, filtered, th.BasicChannel.Id)
+		require.NotContains(t, filtered, "invalid_channel_id")
+	})
+
+	t.Run("should handle empty channel list", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		channels := model.ChannelList{}
+		channelIds := []string{th.BasicChannel.Id}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, channelIds, channels)
+		require.Empty(t, filtered)
+	})
+
+	t.Run("should handle empty channelIds", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		channels := model.ChannelList{
+			th.BasicChannel,
+			th.BasicChannel2,
+		}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, []string{}, channels)
+		require.Empty(t, filtered)
+	})
+
+	t.Run("should handle nil inputs", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, nil, nil)
+		require.Empty(t, filtered)
+	})
+
+	t.Run("should filter duplicate channel IDs", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		channels := model.ChannelList{
+			th.BasicChannel,
+		}
+
+		// Include duplicate channel IDs
+		channelIds := []string{
+			th.BasicChannel.Id,
+			th.BasicChannel.Id,
+		}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, channelIds, channels)
+		require.Len(t, filtered, 2) // Function preserves duplicates as per implementation
+		require.Equal(t, []string{th.BasicChannel.Id, th.BasicChannel.Id}, filtered)
+	})
+
+	t.Run("should log message for filtered channels", func(t *testing.T) {
+		c := &Context{
+			App:        th.App,
+			AppContext: th.Context,
+			Logger:     th.App.Log(),
+		}
+
+		channels := model.ChannelList{
+			th.BasicChannel,
+		}
+
+		channelIds := []string{
+			th.BasicChannel.Id,
+			"invalid_channel_id",
+		}
+
+		filtered := validateSidebarCategoryChannels(c, th.BasicUser.Id, channelIds, channels)
+		require.Len(t, filtered, 1)
+		require.Contains(t, filtered, th.BasicChannel.Id)
+		require.NotContains(t, filtered, "invalid_channel_id")
+	})
+}
+
+func TestDeleteCategoryForTeamForUser(t *testing.T) {
+	th := Setup(t).InitBasic()
+	defer th.TearDown()
+
+	t.Run("should delete category when user has permission", func(t *testing.T) {
```

Comment:
can we add a test that checks once a category is deleted, that the channels it held are moved to the default categories? true for all types of channels.

Thanks